apis/python/tests/test_registration_mappings.py

```diff
@@ -660,59 +723,89 @@ def test_multiples_with_experiment(obs_field_name, var_field_name):
     h5ad3 = create_h5ad_canned(3, obs_field_name, var_field_name)
     h5ad4 = create_h5ad_canned(4, obs_field_name, var_field_name)
 
-    rd = registration.ExperimentAmbientLabelMapping.from_h5ad_appends_on_experiment(
-        experiment_uri=soma1,
-        h5ad_file_names=[h5ad2, h5ad3, h5ad4],
+    rd = tiledbsoma.io.register_h5ads(
+        soma1,
+        [h5ad2, h5ad3, h5ad4],
         measurement_name="measname",
         obs_field_name=obs_field_name,
         var_field_name=var_field_name,
     )
-    assert rd.obs_axis.id_mapping_from_values(["AGAG", "GGAG"]).data == (2, 8)
-    assert rd.var_axes["measname"].id_mapping_from_values(["ESR1", "VEGFA"]).data == (
-        2,
-        4,
+
+    assert_array_equal(
+        rd.obs_axis.id_mapping_from_values(["AGAG", "GGAG"]).data, (2, 8)
+    )
+    assert_array_equal(
+        rd.var_axes["measname"].id_mapping_from_values(["ESR1", "VEGFA"]).data,
+        (
+            2,
+            4,
+        ),
+    )
+    assert_array_equal(
+        rd.var_axes["raw"]
+        .id_mapping_from_values(["ZZZ3", "RAW2", "TP53", "VEGFA"])
+        .data,
+        (9, 6, 3, 4),
     )
-    assert rd.var_axes["raw"].id_mapping_from_values(
-        ["ZZZ3", "RAW2", "TP53", "VEGFA"]
-    ).data == (9, 6, 3, 4)
-
-    assert rd.obs_axis.data == {
-        "AAAT": 0,
-        "ACTG": 1,
-        "AGAG": 2,
-        "CAAT": 3,
-        "CCTG": 4,
-        "CGAG": 5,
-        "GAAT": 6,
-        "GCTG": 7,
-        "GGAG": 8,
-        "TAAT": 9,
-        "TCTG": 10,
-        "TGAG": 11,
-    }
 
-    assert rd.var_axes["measname"].data == {
-        "AKT1": 0,
-        "APOE": 1,
-        "EGFR": 5,
-        "ESR1": 2,
-        "TP53": 3,
-        "VEGFA": 4,
-        "ZZZ3": 6,
-    }
+    assert_frame_equal(
+        rd.obs_axis.joinid_map,
+        pd.Series(
+            {
+                "AAAT": 0,
+                "ACTG": 1,
+                "AGAG": 2,
+                "CAAT": 3,
+                "CCTG": 4,
+                "CGAG": 5,
+                "GAAT": 6,
+                "GCTG": 7,
+                "GGAG": 8,
+                "TAAT": 9,
+                "TCTG": 10,
+                "TGAG": 11,
+            },
+            name="soma_joinid",
+        ).to_frame(),
+        check_names=False,
+    )
 
-    assert rd.var_axes["raw"].data == {
-        "AKT1": 0,
-        "APOE": 1,
-        "ESR1": 2,
-        "TP53": 3,
-        "VEGFA": 4,
-        "RAW1": 5,
-        "RAW2": 6,
-        "EGFR": 7,
-        "RAW3": 8,
-        "ZZZ3": 9,
-    }
+    assert_frame_equal(
+        rd.var_axes["measname"].joinid_map,
+        pd.Series(
+            {
+                "AKT1": 0,
+                "APOE": 1,
+                "ESR1": 2,
+                "TP53": 3,
+                "VEGFA": 4,
+                "EGFR": 5,
+                "ZZZ3": 6,
+            },
+            name="soma_joinid",
+        ).to_frame(),
+        check_names=False,
+    )
+
+    assert_frame_equal(
```

Comment:
`assert_frame_equal` and `assert_array_equal` are awesome, and we should use them more often.

Switching to them is not a crucial part of this PR, and, ideally, would be a separate PR -- and, doing so would reduce the line-count on this large and impactful PR. I believe you're rebase-averse, so I won't insist on you splitting these out, but as a general rule I do ask people to split out reworkings/renamings from functional changes as it reduces review time and PR complexity.

Reply:
noted.  will leave them in.